### PR TITLE
Diagrams for integrations

### PR DESCRIPTION
**Pull request by mooreds**
These are general diagrams about flows for four use cases:
* the hosted login pages oauth grant
* the login api
* the login api with federation/complete login
* the hosted login pages with federation and the oauth grant

I also did some cleanup on the identity provider pages to make them less verbose.

### CODE REVIEW

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/gaming/xbox.mdx [hunk 1]
```diff
@@ -28,11 +29,11 @@ Once you have completed this configuration you will be able to enable the Xbox l
 
 <img src="/img/docs/lifecycle/authenticate-users/identity-providers/gaming/login-xbox.png" alt="Xbox Login" role="bottom-cropped" width="1200" />
 
-## Create an Xbox Account
+<IdentityProviderOverviewDiagram identity_provider_name="XBox" />
```
Comment (mrudatsprint) [inline astro/src/content/docs/lifecycle/authenticate-users/identity-providers/gaming/xbox.mdx:32]: I'm being totally picky and should `XBox` be `Xbox`?

File: astro/src/diagrams/docs/get-started/core-concepts/integration-points/login-api-flow-federation.astro [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
```
Comment (mrudatsprint) [inline astro/src/diagrams/docs/get-started/core-concepts/integration-points/login-api-flow-federation.astro:1]: Pretty cool how a sequence diagram can be created.

**Review by mrudatsprint — APPROVED** [review]

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/gaming/xbox.mdx (hunk 1 above)
Comment (mooreds) [reply astro/src/content/docs/lifecycle/authenticate-users/identity-providers/gaming/xbox.mdx:32]: Great catch. Looks like we use `Xbox` for our docs, so let me correct that.

File: astro/src/diagrams/docs/get-started/core-concepts/integration-points/login-api-flow-federation.astro (hunk 2 above)
Comment (mooreds) [reply astro/src/diagrams/docs/get-started/core-concepts/integration-points/login-api-flow-federation.astro:1]: 100%

File: astro/src/diagrams/docs/get-started/core-concepts/integration-points/oauth-grant-with-federation.astro [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    participant FusionAuth
+    participant IdentityProvider
+
+    User ->> App : View Initial Page<br/>Click Login
+    App ->> User : Redirect To <br/>FusionAuth Authorization URL
+    User ->> FusionAuth : Request Login Page
+    FusionAuth ->> User : Return Login Page
+    User ->> FusionAuth : Clicks On 'Login With Identity Provider'
+    FusionAuth ->> User : Redirect To Identity Provider Authorization URL
+    IdentityProvider ->> IdentityProvider : Validate Credentials
```
Comment (bhalsey) [inline astro/src/diagrams/docs/get-started/core-concepts/integration-points/oauth-grant-with-federation.astro:19]: Before `IdentityProvider: Validate Credentials`, we should have something like
`User ->> IdentityProvider : Enters Credentials`

File: astro/src/content/docs/lifecycle/authenticate-users/identity-providers/social/twitter.mdx [hunk 4]
```diff
@@ -21,11 +22,11 @@ Once you have completed this configuration you will be able to enable the Twitte
...
 <img src="/img/docs/lifecycle/authenticate-users/identity-providers/social/login-twitter.png" alt="Twitter Login" width="1200"  role="bottom-cropped" />
 
-## Create a Twitter Account
+<IdentityProviderOverviewDiagram identity_provider_name="Twitter" />
 
-Ensure you have a Twitter account, and then navigate to the Twitter Developer dashboard.
+## Create a Twitter Account
 
-https://developer.twitter.com
+Ensure you have a Twitter account, and then navigate to the [Twitter Developer dashboard](https://developer.twitter.com).
```
Comment (bhalsey) [inline astro/src/content/docs/lifecycle/authenticate-users/identity-providers/social/twitter.mdx:29]:
```suggestion
Ensure you have a Twitter account, and then navigate to the [Twitter Developer Portal](https://developer.twitter.com).
```

**Pull request merged**